### PR TITLE
Add option to enforce identical app name in pings

### PR DESCRIPTION
**Pull request by yulunli**
Always reject pings from wrong app name

### CODE REVIEW

File: server/protocol/ping.js [hunk 1]
```diff
@@ -40,6 +40,18 @@ module.exports = function createPingHandler(ringpop) {
             return callback(new Error('need req body with source, changes, and checksum'));
         }
 
+        if (body.hasOwnProperty('app')) {
+            if (body.app !== ringpop.app) {
+                ringpop.logger.warn('Rejected ping from wrong ringpop app', body);
```
Comment (mennopruijssers) [inline server/protocol/ping.js]: log as error?

File: server/protocol/ping.js [hunk 2]
```diff
@@ -40,6 +40,18 @@ module.exports = function createPingHandler(ringpop) {
...
+                return callback(new Error('Pinged ringpop has a different app name'));
+            }
+        } else {
+            if (ringpop.requiresAppInPing) {
+                ringpop.logger.warn('Rejected ping from unknown ringpop app', body);
```
Comment (mennopruijssers) [inline server/protocol/ping.js]: same?

**Comment by mennopruijssers** [comment]
Oh, I just saw ringpop-common#97. Can you add the integration test to that one and temporary change the branch name in [run-shared-integration-tests:9](https://github.com/uber-node/ringpop-node/blob/master/test/run-shared-integration-tests#L9) temporary until ringpop-common#97 is merged

File: server/protocol/ping.js (hunk 1 above)
Comment (yulunli) [inline server/protocol/ping.js]: I think warning is better because if another service is pinging incorrectly, it's not an error for the pinged service.

**Comment by proflayton** [comment]
this lgtm

**Comment by yulunli** [comment]
Given we already have a test in ringpop-node, the test in ringpop-common will be added at a later time.

**Pull request merged**